### PR TITLE
1.6.1: allow configuration of admin user via env vars

### PR DESCRIPTION
**Pull request by klaemo**
feedback welcome @kxepal @yosifkit

### CODE REVIEW

File: 1.6.1/docker-entrypoint.sh [hunk 1]
```diff
@@ -14,22 +14,52 @@
...
-  chmod 775 /usr/local/etc/couchdb/*.d
-  exec gosu couchdb "$@"
+	# Start CouchDB
+	couchdb -b > /dev/null 2>&1
+	while ! curl -s 127.0.0.1:5984 > /dev/null; do sleep 1; done
+
+	if [ "$COUCHDB_USER" ] && [ "$COUCHDB_PASSWORD" ]; then
+		# Create user
+		echo "Creating CouchDB admin: \"$COUCHDB_USER\"..."
+		curl -X PUT -sS 127.0.0.1:5984/_config/admins/$COUCHDB_USER -d \"${COUCHDB_PASSWORD}\" > /dev/null 2>&1
```
Comment (kxepal) [inline 1.6.1/docker-entrypoint.sh]: Alternatively, you can add your own, say, `docker.ini` file where admin credentials will be written. On first start CouchDB will replace password with the hash.

Comment (klaemo) [inline 1.6.1/docker-entrypoint.sh]: Yeah, I thought of that too. That would eliminate the need to start couch just for PUTting the admin, but we couldn't display the warning below as easily.

Comment (kxepal) [inline 1.6.1/docker-entrypoint.sh]: Idea: check if env variables provided. If they does - no need in warning. Is it possible to provide own ini files for docker image? If so, is it possible to grep it for admins section? If yes and such was found - unlikely there will be admin party.

File: 1.6.1/docker-entrypoint.sh [hunk 2]
```diff
@@ -14,22 +14,52 @@
...
+	# Start CouchDB
+	couchdb -b > /dev/null 2>&1
+	while ! curl -s 127.0.0.1:5984 > /dev/null; do sleep 1; done
+
+	if [ "$(curl -s 127.0.0.1:5984/_config/admins)" == "{}" ]; then
+		# The - option suppresses leading tabs but *not* spaces. :)
+		cat >&2 <<-'EOWARN'
+			****************************************************
+			WARNING: CouchDB is running in Admin Party mode.
+							 This will allow anyone with access to the
```
Comment (josegonzalez) [inline 1.6.1/docker-entrypoint.sh]: Should there be all this extra whitespace?

Comment (klaemo) [inline 1.6.1/docker-entrypoint.sh]: no probably not :)

**Pull request merged**